fe/fe-core/src/main/java/com/starrocks/connector/jdbc/JDBCMetadata.java

```diff
@@ -76,11 +74,9 @@ public Connection getConnection() throws SQLException {
 
     @Override
     public List<String> listDbNames() {
-        try (Connection connection = getConnection()) {
-            return Lists.newArrayList(schemaResolver.listSchemas(connection));
-        } catch (SQLException e) {
-            throw new StarRocksConnectorException(e.getMessage());
-        }
+        ImmutableMap<String, String> metaInfo =
+                ImmutableMap.of("func", "listDbNames");
+        return catalogCache.get(metaInfo, k -> metaResolver.refreshCacheForListDbNames());
```

Comment:
as async cache, if it doesn't hit on cache, it will return null? if you need wait the result why not use sync cache?